alu_u32/src/lib.rs

```diff
@@ -3,6 +3,7 @@
 const ADD32_OPCODE: u32 = 8;
 const SUB32_OPCODE: u32 = 9;
 const MUL32_OPCODE: u32 = 10;
+#[allow(dead_code)]
```

Comment:
I worry suppressing unused warnings may lead to forgotten dead code. I think arguably the "right" approach is to leave the warning until they're really fixed, but I know it's also nice to have a warning-free main branch, so maybe add TODOs to make sure dead code isn't forgotten?

Reply:
I didn't really know why you have this unused code.

You are right that a TODO might be useful. Though I always just treat such a dead-code annotation as an implicit TODO.

The best alternative is perhaps to just delete unused code and resurrect it from hit history when needed?

Reply:
I think in this case temporarily deleting `LT_OPCODE` seems fine, since it's easy to restore one line. But I think it can sometimes be convenient to commit internal functions before the publicly-exported API is ready. Like thinking about top-down vs bottom-up development, a no-dead-code policy would require committing code in a top-down order, which might be less natural for some authors. Probably makes sense for more mature codebases though.

I guess treating `allow(dead_code)` / `allow(unused)` as implicit TODOs also seems fine.